### ISSUE TITLE
Проверить работу gdcc в терминальном режиме

### ISSUE THREAD

**Issue by gsbelarus** [GoldenSoftwareLtd/gedemin#4009]
Если на терминальном сервере несколько пользователей одновременно работают с гедымином, то, говорят, что gdcc отображает прогресс действий только одного из них.

**Comment by AliakseiDanilchyk** [comment]
Точнее только у одного (скорее всего первого подключившегося) пользователя gdcc работает. При этом показывает логи и прогресс-бары и от всех Гедыминов запущенных под другими пользователями. У остальных пользователей gdcc запускается но не показывает подключения к Гедымину в сеансе этого пользователя.

**Comment by gsbelarus** [comment]
Да. Проблема в обработке WM_GDCC_CONNECT в методе TgdccClient.ProcessMessage.

Надо:

1. Проверить не в терминальной ли мы сессии
2. Если да, то сначала пытаемся связаться с gdcc по номеру порта. Возможно он уже запущен в другой терминальной сессии. Для проверки надо будет реализовать команду ping в нашем протоколе и ответ на нее сервера.
3. Если запущен, то надо брать следующий номер порта пока не найдем свободный.
4. запускаем утилиту gdcc.exe передавая ей через параметры командной строки найденный доступный номер порта.
5. дальше как обычно, устанавливаем связь и работаем.
6. если не в терминальной сессии, то действуем как раньше.
